core/app/models/spree/taxon.rb

```diff
@@ -21,6 +21,7 @@ class Taxon < Spree::Base
     validates :meta_keywords, length: { maximum: 255 }
     validates :meta_description, length: { maximum: 255 }
     validates :meta_title, length: { maximum: 255 }
+    validates :taxonomy_id, uniqueness: { message: 'can only have one root Taxon' }, if: :root?
```

Comment:
It would be better to rely on I18n for this so that it can be set separately per language for stores that need that. https://guides.rubyonrails.org/i18n.html#error-message-scopes